Storybook is not compatible with libraries they have only library-secondary-entry-point

### Steps to reproduce:

1: Create a angular:library without a module
```
nx generate @nrwl/angular:library test --buildable --compilationMode=partial --importPath=@testorg/test --prefix=test --publishable --skipModule --standaloneConfig
```

2: Generate some library-secondary-entry-point
```
nx generate @nrwl/angular:library-secondary-entry-point header --library=test
nx generate @nrwl/angular:library-secondary-entry-point footer --library=test
```

3: Generatet some angular:component for the generated Libs
(Why i must define path here? why the header.module can not be found automaticly?)
```
nx generate @nrwl/angular:component header --project=test --module=header.module --path=/libs/test/header/src/lib
nx generate @nrwl/angular:component header-menu --project=test --module=header.module --path=/libs/test/header/src/lib
nx generate @nrwl/angular:component footer --project=test --module=footer.module --path=/libs/test/footer/src/lib
nx generate @nrwl/angular:component footer-menu --project=test --module=footer.module --path=/libs/test/footer/src/lib
```

4: Add storybook-configuration
```
nx generate @nrwl/angular:storybook-configuration test --tsConfiguration
```

Now Execute `nx run test:storybook` and you will get this error:

```bash
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (implicitDependencies) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name ($schema) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (implicitDependencies) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (configFilePath) found.
Project extension with invalid name ($schema) found.
Project extension with invalid name (tags) found.
Project extension with invalid name (implicitDependencies) found.
Project extension with invalid name (configFilePath) found.
info => Loading presets
info Addon-docs: using MDX1
info => Using implicit CSS loaders
info => Loading angular-cli config for angular >= 13.0.0
info => Using angular browser target options from "test:build-storybook"
info => Using angular project with "tsConfig:/home/budi/AnimeSector/libs/test/.storybook/tsconfig.json"
info => Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
6% setup before compile angular-compiler<i> [webpack-dev-middleware] wait until bundle finished
8% setup compilation angular-compilerError: ENOENT: no such file or directory, open '/home/budi/AnimeSector/node_modules/@nrwl/react/typings/styled-jsx.d.ts'
    at Object.openSync (node:fs:585:3)
    at Module.readFileSync (node:fs:453:35)
    at NodeJSFileSystem.readFile (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-TOKOIIBI.js:333:16)
    at EsmDependencyHost.recursivelyCollectDependencies (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:217:34)
    at EsmDependencyHost.processFile (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:246:12)
    at EsmDependencyHost.collectDependenciesInFiles (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:213:12)
    at ProgramBasedEntryPointFinder.getInitialEntryPointPaths (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:864:10)
    at ProgramBasedEntryPointFinder.findEntryPoints (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:823:35)
    at file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:1273:33
    at new ClusterMaster (file:///home/budi/AnimeSector/node_modules/@angular/compiler-cli/bundles/chunk-MN3YDSGE.js:1375:22)
19% building 2/13 entries 4780/4800 dependencies 908/1107 modulesassets by chunk 5.07 MiB (id hint: vendors)
  assets by status 4.85 MiB [big]
    asset vendors-node_modules_storybook_addon-actions_manager_js-node_modules_storybook_addon-backgrou-45da3d.manager.bundle.js 3.42 MiB [emitted] [big] (id hint: vendors)
    asset vendors-node_modules_storybook_components_dist_esm_formatter-0d5cb0eb_js.manager.bundle.js 886 KiB [emitted] [big] (id hint: vendors)
    asset vendors-node_modules_storybook_components_dist_esm_OverlayScrollbars-1355f44c_js.manager.bundle.js 295 KiB [emitted] [big] (id hint: vendors)
    asset vendors-node_modules_storybook_components_dist_esm_syntaxhighlighter-b07b042a_js.manager.bundle.js 286 KiB [emitted] [big] (id hint: vendors)
  asset vendors-node_modules_storybook_components_dist_esm_WithTooltip-167e9982_js.manager.bundle.js 122 KiB [emitted] (id hint: vendors)
  asset vendors-node_modules_storybook_components_dist_esm_Color-f953d088_js.manager.bundle.js 82.8 KiB [emitted] (id hint: vendors)
  asset vendors-node_modules_storybook_components_dist_esm_GlobalScrollAreaStyles-8793ce4a_js.manager.bundle.js 15.5 KiB [emitted] (id hint: vendors)
asset runtime~main.manager.bundle.js 14.2 KiB [emitted] (name: runtime~main)
asset index.html 2.79 KiB [emitted]
asset main.manager.bundle.js 1.37 KiB [emitted] (name: main)
asset node_modules_unfetch_dist_unfetch_js.manager.bundle.js 1.28 KiB [emitted]
Entrypoint main [big] 3.44 MiB = runtime~main.manager.bundle.js 14.2 KiB vendors-node_modules_storybook_addon-actions_manager_js-node_modules_storybook_addon-backgrou-45da3d.manager.bundle.js 3.42 MiB main.manager.bundle.js 1.37 KiB
orphan modules 973 KiB [orphan] 94 modules
runtime modules 8.72 KiB 15 modules
modules by path ./node_modules/ 4.47 MiB 857 modules
./util.inspect (ignored) 15 bytes [built] [code generated]

WARNING in DefinePlugin
Conflicting values for 'process.env.NODE_ENV'

1 WARNING in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)

1 warning has detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

manager (webpack 5.73.0) compiled with 2 warnings in 15055 ms
99% done plugins webpack-hot-middlewarewebpack built preview 11ca998f3a5d3545e3bd in 17387ms
WARN Force closed manager build
ModuleBuildError: Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
Error: Emit attempted before Angular Webpack plugin initialization.
    at /home/budi/AnimeSector/node_modules/@ngtools/webpack/src/ivy/loader.js:81:18
    at processResult (/home/budi/AnimeSector/node_modules/webpack/lib/NormalModule.js:758:19)
    at /home/budi/AnimeSector/node_modules/webpack/lib/NormalModule.js:860:5
    at /home/budi/AnimeSector/node_modules/loader-runner/lib/LoaderRunner.js:400:11
    at /home/budi/AnimeSector/node_modules/loader-runner/lib/LoaderRunner.js:252:18
    at context.callback (/home/budi/AnimeSector/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at /home/budi/AnimeSector/node_modules/@ngtools/webpack/src/ivy/loader.js:81:9
ModuleNotFoundError: Module not found: Error: Can't resolve './libs/test/src/lib' in '/home/budi/AnimeSector'
    at /home/budi/AnimeSector/node_modules/webpack/lib/Compilation.js:2016:28
    at /home/budi/AnimeSector/node_modules/webpack/lib/ContextModuleFactory.js:214:15
    at /home/budi/AnimeSector/node_modules/neo-async/async.js:2830:7
    at /home/budi/AnimeSector/node_modules/neo-async/async.js:6877:13
    at /home/budi/AnimeSector/node_modules/webpack/lib/ContextModuleFactory.js:184:26
    at finishWithoutResolve (/home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:312:11)
    at /home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:386:15
    at /home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/home/budi/AnimeSector/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /home/budi/AnimeSector/node_modules/webpack/lib/cache/ResolverCachePlugin.js:257:15
ModuleNotFoundError: Module not found: Error: Can't resolve './libs/test/src/lib' in '/home/budi/AnimeSector'
    at /home/budi/AnimeSector/node_modules/webpack/lib/Compilation.js:2016:28
    at /home/budi/AnimeSector/node_modules/webpack/lib/ContextModuleFactory.js:214:15
    at /home/budi/AnimeSector/node_modules/neo-async/async.js:2830:7
    at /home/budi/AnimeSector/node_modules/neo-async/async.js:6877:13
    at /home/budi/AnimeSector/node_modules/webpack/lib/ContextModuleFactory.js:184:26
    at finishWithoutResolve (/home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:312:11)
    at /home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:386:15
    at /home/budi/AnimeSector/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/home/budi/AnimeSector/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /home/budi/AnimeSector/node_modules/webpack/lib/cache/ResolverCachePlugin.js:257:15
Error: Failed to initialize Angular compilation - NGCC failed.
    at /home/budi/AnimeSector/node_modules/@ngtools/webpack/src/ivy/plugin.js:134:41
    at Hook.eval (eval at create (/home/budi/AnimeSector/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:14:1)
    at Hook.CALL_DELEGATE [as _call] (/home/budi/AnimeSector/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/home/budi/AnimeSector/node_modules/webpack/lib/Compiler.js:1121:30)
    at /home/budi/AnimeSector/node_modules/webpack/lib/Compiler.js:1166:29
    at _next0 (eval at create (/home/budi/AnimeSector/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
    at eval (eval at create (/home/budi/AnimeSector/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:55:1)


 >  NX   Broken build, fix the error above.

   You may need to refresh the browser.


 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target storybook for project test (21s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

I think this happens because storybook doesn't look for a library-secondary-entry-point, but only for a normal library. however, very few people want a library where you can't import each package individually.

```bash
 >  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : linux x64
   npm  : 8.16.0
   
   nx : 14.5.2
   @nrwl/angular : 14.5.1
   @nrwl/cypress : 14.5.1
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.5.1
   @nrwl/eslint-plugin-nx : 14.5.1
   @nrwl/express : Not Found
   @nrwl/jest : 14.5.1
   @nrwl/js : 14.5.1
   @nrwl/linter : 14.5.1
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.5.2
   @nrwl/web : 14.5.1
   @nrwl/workspace : 14.5.1
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
         @ionic/angular: 6.2.1
         @ngneat/transloco: 3.2.0
         @nxtend/capacitor: 13.0.0
         @nxtend/ionic-angular: 13.1.0
         @storybook/angular: 6.5.10
```

Can relate. We have only secondary-entry-points for better code-splitting etc.  

shared/ui
  - inputs
  - buttons
  - etc..
  
`  ModuleNotFoundError: Module not found: Error: Can't resolve './libs/portal/shared/ui/src/lib' in `

--- a/packages/storybook/src/generators/configuration/project-files-ts/.storybook/main.ts__tmpl__
+++ b/packages/storybook/src/generators/configuration/project-files-ts/.storybook/main.ts__tmpl__
@@ -9,10 +9,12 @@ const config: StorybookConfig = {
   <% } %>
 
   stories: [
-    ...rootMain.stories,
+    ...rootMain.stories,<% if(uiFramework === '@storybook/angular' && projectType === 'library') { %>
+    '../**/*.stories.mdx',
+    '../**/*.stories.@(js|jsx|ts|tsx)' <% } else { %>
     '../<%= projectDirectory %>/**/*.stories.mdx',
     '../<%= projectDirectory %>/**/*.stories.@(js|jsx|ts|tsx)'
-  ],
+  <% } %>],
   addons: [...(rootMain.addons || []) <% if(uiFramework === '@storybook/react') { %>, '@nrwl/react/plugins/storybook' <% } %><% if(uiFramework === '@storybook/react-native') { %>, '@storybook/addon-ondevice-actions',  '@storybook/addon-ondevice-backgrounds', '@storybook/addon-ondevice-controls', '@storybook/addon-ondevice-notes'  <% } %>
     <% if(usesSwc && !isNextJs) { %>, 'storybook-addon-swc' <% } %>
     <% if(isNextJs) { %>, 'storybook-addon-swc', 

--- a/packages/storybook/src/generators/configuration/project-files-ts/.storybook/tsconfig.json__tmpl__
+++ b/packages/storybook/src/generators/configuration/project-files-ts/.storybook/tsconfig.json__tmpl__
@@ -11,12 +11,19 @@
   ],<% } %>
   <% if(uiFramework === '@storybook/angular' && projectType === 'application') { %>"files": ["../src/polyfills.ts"],<% } %>
   "exclude": ["../**/*.spec.ts" <% if(uiFramework === '@storybook/react') { %>, "../**/*.spec.js", "../**/*.spec.tsx", "../**/*.spec.jsx"<% } %>],
-  "include": [
-    "../<%= mainDir %>/**/*.stories.ts", 
-    "../<%= mainDir %>/**/*.stories.js", 
-    "../<%= mainDir %>/**/*.stories.jsx", 
-    "../<%= mainDir %>/**/*.stories.tsx", 
-    "../<%= mainDir %>/**/*.stories.mdx", 
+  "include": [<% if(uiFramework === '@storybook/angular' && projectType === 'library') { %>
+    "../**/*.stories.ts",
+    "../**/*.stories.js",
+    "../**/*.stories.jsx",
+    "../**/*.stories.tsx",
+    "../**/*.stories.mdx",
     "*.ts",
-    "*.js"<% if(uiFramework === '@storybook/react-native') { %>, "*.tsx"<% } %>]
+    "*.js"<% } else { %>
+    "../<%= mainDir %>/**/*.stories.ts",
+    "../<%= mainDir %>/**/*.stories.js",
+    "../<%= mainDir %>/**/*.stories.jsx",
+    "../<%= mainDir %>/**/*.stories.tsx",
+    "../<%= mainDir %>/**/*.stories.mdx",
+    "*.ts",
+    "*.js"<% if(uiFramework === '@storybook/react-native') { %>, *.tsx"<% } %><% } %>]
 }

--- a/packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__
+++ b/packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__
@@ -8,10 +8,12 @@ module.exports = {
   core: { ...rootMain.core, builder: 'webpack5' },
   <% } %>
   stories: [
-    ...rootMain.stories,
+    ...rootMain.stories,<% if(uiFramework === '@storybook/angular' && projectType === 'library') { %>
+    '../**/*.stories.mdx',
+    '../**/*.stories.@(js|jsx|ts|tsx)' <% } else { %>
     '../<%= projectDirectory %>/**/*.stories.mdx',
     '../<%= projectDirectory %>/**/*.stories.@(js|jsx|ts|tsx)'
-  ],
+  <% } %>],
   addons: [...rootMain.addons <% if(uiFramework === '@storybook/react') { %>, '@nrwl/react/plugins/storybook' <% } %><% if(uiFramework === '@storybook/react-native') { %>, '@storybook/addon-ondevice-actions',  '@storybook/addon-ondevice-backgrounds', '@storybook/addon-ondevice-controls', '@storybook/addon-ondevice-notes'  <% } %>
     <% if(usesSwc && !isNextJs) { %>, 'storybook-addon-swc' <% } %>
     <% if(isNextJs) { %>, 'storybook-addon-swc', 

--- a/packages/storybook/src/generators/configuration/project-files/.storybook/tsconfig.json__tmpl__
+++ b/packages/storybook/src/generators/configuration/project-files/.storybook/tsconfig.json__tmpl__
@@ -11,11 +11,17 @@
   ],<% } %>
   <% if(uiFramework === '@storybook/angular' && projectType === 'application') { %>"files": ["../src/polyfills.ts"],<% } %>
   "exclude": ["../**/*.spec.ts" <% if(uiFramework === '@storybook/react') { %>, "../**/*.spec.js", "../**/*.spec.tsx", "../**/*.spec.jsx"<% } %>],
-  "include": [
+  "include": [<% if(uiFramework === '@storybook/angular' && projectType === 'library') { %>
+    "../**/*.stories.ts",
+    "../**/*.stories.js",
+    "../**/*.stories.jsx",
+    "../**/*.stories.tsx",
+    "../**/*.stories.mdx",
+    "*.js"<% } else { %>
     "../<%= mainDir %>/**/*.stories.ts", 
     "../<%= mainDir %>/**/*.stories.js", 
     "../<%= mainDir %>/**/*.stories.jsx", 
     "../<%= mainDir %>/**/*.stories.tsx", 
-    "../<%= mainDir %>/**/*.stories.mdx", 
-    "*.js"<% if(uiFramework === '@storybook/react-native') { %>, "*.ts", "*.tsx"<% } %>]
+    "../<%= mainDir %>/**/*.stories.mdx",
+    "*.js"<% if(uiFramework === '@storybook/react-native') { %>, "*.ts", "*.tsx"<% } %><% } %>]
 }

